packages/kg-default-nodes/lib/nodes/header/HeaderNode.js

```diff
@@ -0,0 +1,269 @@
+import {KoenigDecoratorNode} from '../../KoenigDecoratorNode';
+import {renderHeaderNodeToDOM} from './HeaderRenderer';
+import {HeaderParser} from './HeaderParser';
+import {createCommand} from 'lexical';
+
+export const INSERT_HEADER_COMMAND = createCommand();
+
+export class HeaderNode extends KoenigDecoratorNode {
+    // header payload properties
+
+    __size;
+    __style;
+    __buttonEnabled;
+    __buttonUrl;
+    __buttonText;
+    __header;
+    __subheader;
+    __hasHeader;
+    __hasSubheader;
+    __backgroundImageStyle;
+    __backgroundImageSrc;
+
+    static getType() {
+        return 'header';
+    }
+
+    static clone(node) {
+        return new this(
+            node.getDataset(),
+            node.__key
+        );
+    }
+
+    static get urlTransformMap() {
+        return {
+            buttonUrl: 'url',
+            backgroundImageSrc: 'url'
```

Comment:
Both header and subheader also support inline links
```suggestion
            buttonUrl: 'url',
            backgroundImageSrc: 'url',
            header: 'html',
            subheader: 'html'
```